app/javascript/vue/repository_item_sidebar/repository_values/ScrollSpy.vue

```diff
@@ -35,176 +31,141 @@ export default {
+        this.setActiveSection(this.sections[0]);
         return;
       }
 
-      // Far bottom position
-      if (this.bodyContainerEl.scrollTop + this.bodyContainerEl.clientHeight === this.bodyContainerEl.scrollHeight) {
-        this.shouldRecalculateWhenStopped = false
-        this.handleTopOrBotScrollPosition(this.sections[this.sections.length - 1])
-        return
+      // Determine the scroll direction (down or up)
+      if (currentScrollTop > this.previousScrollTop) {
+        this.handleScrollDown();
+      } else {
+        this.handleScrollUp();
       }
 
-      // Checks when a section enters targetArea's boundary and highlights it
-      for (const section of this.sections) {
-        const sectionRect = section.getBoundingClientRect();
-        if (sectionRect === this.prevSection) continue;
+      this.previousScrollTop = currentScrollTop;
+    },
+
+    // scrolling from up -> down
+    // highlighting based on proximity to the header
+    handleScrollDown() {
+      const headerTop = this.getDistanceToTop(this.header);
+      const nearestSection = this.sections.reduce((acc, section) => {
+        const distance = Math.abs(headerTop - this.getDistanceToTop(section));
+        return distance < this.proximityThreshold ? section : acc;
+      }, null);
 
-        if (this.isSectionInBounds(sectionRect, bodyWrapperTargetAreaRect, margin)) {
-          this.handleSectionHighlight(section);
-        }
-      }
+      if (nearestSection) this.setActiveSection(nearestSection);
     },
 
-    // For handling top/bottom most positions
-    handleTopOrBotScrollPosition(section) {
-      const sectionId = section.getAttribute('id');
-      const foundObj = this.itemsToCreate.find((obj) => obj?.sectionId === sectionId);
+    // scrolling from down -> up
+    // highlighting based on passing out of view
+    handleScrollUp() {
+      if (!this.activeSection) return;
 
-      if (foundObj) {
-        this.selectedNavText = foundObj.textId;
-        this.selectedNavIndicator = foundObj.id;
+      const activeSectionRect = this.activeSection.getBoundingClientRect();
+      const containerRect = this.bodyContainerEl.getBoundingClientRect();
+
+      if (activeSectionRect.bottom < containerRect.top || activeSectionRect.top > containerRect.bottom) {
+        const previousSection = this.getPreviousSection(this.activeSection);
+        if (previousSection) this.setActiveSection(previousSection);
       }
     },
 
-    // For checking if a section is within targetArea's boundaries
-    isSectionInBounds(sectionRect, targetAreaRect, margin) {
-      const upperBound = targetAreaRect.top - margin;
-      const lowerBound = targetAreaRect.top + margin;
-      return sectionRect.top >= upperBound && sectionRect.top <= lowerBound;
+    getPreviousSection(currentSection) {
+      const index = this.sections.indexOf(currentSection);
+      return index > 0 ? this.sections[index - 1] : null;
+    },
+
+    getDistanceToTop(el) {
```

Comment:
this method may be a bit unnecessary, you could just call `getBoundingClientRect().top` directly